lib/SILGen/SILGenApply.cpp

```diff
@@ -4549,12 +4585,34 @@ RValue SILGenFunction::emitApply(ResultPlanPtr &&resultPlan,
     emitForeignErrorCheck(loc, directResults, errorTemp, doesNotThrow,
                           *foreignError);
   }
-  
+
+  SmallVector<CleanupHandle, 8> cleanupHandles;
+  // For objc async calls, push cleanup to be used on throw paths in the result
+  // planner.
+  if (calleeTypeInfo.foreign.async) {
+    for (auto unmanagedCopy : unmanagedCopies) {
+      Cleanups.pushCleanup<FixLifetimeDestroyCleanup>(unmanagedCopy.getValue());
```

Comment:
Rather than use a new array, you can just reuse the ManagedValue array:

```
for (unsigned i : indices(unmanagedCopies)) {
    SILValue value = unmanagedCopies[i].getValue();
    Cleanups.pushCleanup<FixLifetimeDestroyCleanup>(value);
    auto handle = Cleanups.getTopCleanup();
    unmanagedCopies[i] = ManagedValue(value, cleanup);
}
```

Reply:
Nice! thanks.